File: linked-lists/LinkedList.java
class LinkedList {
  public static void main(String[] args) {
    // create linked list
    Node linkedList = new Node(5, null);
    // add to linked list
    linkedList = new Node(-2, linkedList);
    linkedList = new Node(3, linkedList);

    // print out linked list
    while(linkedList != null) {
      System.out.println(linkedList.data);
      linkedList = linkedList.next;
    }
  }
}

class Node {
  int data;
  Node next;

  Node(int data, Node next) {
    this.data = data;
    this.next = next;
  }
}

File: linked-lists/linked_list.cpp
#include <iostream>

using namespace std;

struct node {
  int data;
  node* next;
  
  node(int data, node* next) {
    this->data = data;
    this->next = next;
  }
};

int main() {
  // create linked list
  node* linked_list = new node(2, NULL);
  // add to linked list
  linked_list = new node(-1, linked_list);
  linked_list = new node(7, linked_list);

 vector<int>v;
  // print out linked list
  while(linked_list != NULL) {
    cout << linked_list->data << endl;
    v.push_back(linked_list->data); // storing elements in an vector for printing in reverse manner since we cannot traverse in reverse direction in singly linked list. 
    linked_list = linked_list->next;
  }
  
  //Printing output in reverse order
  for (auto ir = v.rbegin(); ir != v.rend(); ++ir) 
        cout << *ir << " "; 
  return 0;
  
}
